gldcore/object.c

```diff
@@ -1212,7 +1212,7 @@ OBJECT *object_get_next(OBJECT *obj){ /**< the object from which to start */
 	the request to prevent looping.  This will prevent
 	an object_set_parent call from creating a parent loop.
  */
-static int _set_rank(OBJECT *obj, OBJECTRANK rank, OBJECT *first)
+static unsigned int _set_rank(OBJECT *obj, OBJECTRANK rank, OBJECT *first)
```

Comment:
So OBJECTRANK is an unsigned and this makes sense, but will the various `return -1;` error conditions cause any issues with this?

Reply:
Strictly speaking, casting -1 to unsigned int will produce UINT_MAX, which would be a valid value. I would classify this as an altogether separate bug. 
our options are probably to change the error detection value to be UINT_MAX, or change OBJECTRANK to be signed.

Reply:
Fair enough.  As long as we never have UINT_MAX objects/ranks, it shouldn't have any ambiguity.